src/validation/tests/variable_validation_tests.rs

```diff
@@ -360,3 +362,17 @@ mod overflows {
         assert_validation_snapshot!(diagnostics);
     }
 }
+
+#[test]
+fn type_initializers_in_structs_are_validated() {
+    let diagnostics = parse_and_validate_buffered(
+        "
+        TYPE MyStruct: STRUCT
+            overflow : SINT := 1234567890;
+            invalid_array_assignment : ARRAY[0..1] OF INT := 0;
+        END_STRUCT END_TYPE
```

Comment:
This is a very minimal test and I'd like to add a few more cases here but unfortunately it seems some validations such as `x : INT := 'incorrect type'` are only validated in the codegen - we should identify such cases and try to migrate them to the validation stage if possible.

Reply:
You're right, thanks!